File: app/src/main/java/io/github/ovso/psytest/ui/main/MainPresenter.java
package io.github.ovso.psytest.ui.main;

import android.arch.lifecycle.Lifecycle;
import android.arch.lifecycle.LifecycleObserver;
import android.arch.lifecycle.OnLifecycleEvent;

public interface MainPresenter extends LifecycleObserver {

  @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
  void onCreate();

  interface View {
    void setupView();

    void setupTabLayout();

    void setupViewPager();

    void showTabName(int position, String name);

    void refresh();

    void showTitle(String title);

    void showBannerAd();

    void changeTheme();
  }
}

File: app/src/main/java/io/github/ovso/psytest/ui/main/MainPresenter.kt
package io.github.ovso.psytest.ui.main

import android.arch.lifecycle.Lifecycle
import android.arch.lifecycle.LifecycleObserver
import android.arch.lifecycle.OnLifecycleEvent
import io.github.ovso.psytest.ui.main.adapter.MainItem

interface MainPresenter : LifecycleObserver {

  @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
  fun onCreate()

  interface View {
    fun setupView()
    fun showTitle(title: String)
    fun setupRv()
    fun loadAd()
    fun submitList(it: List<MainItem>)
  }
}
